pkg/nodeconfig/nodeconfig.go

```
@@ -346,13 +346,17 @@ func (nc *nodeConfig) createFilesFromIgnition() (map[string]string, error) {
 		return nil, err
 	}
 
-	filesToTransfer := map[string]struct{}{
-		ignition.KubeletCACertPath: {},
-	}
+	filesToTransfer := map[string]struct{}{}
 	if _, ok := kubeletArgs[ignition.CloudConfigOption]; ok {
 		filesToTransfer[ignition.CloudConfigPath] = struct{}{}
 	}
 	filePathsToContents := make(map[string]string)
+	// process kubelet-ca
+	filePathsToContents[windows.K8sDir+"\\"+KubeletClientCAFilename] = string(ign.GetKubeletCAData())
```

if `ign.GetKubeletCAData()` returns an empty string, the file will be created with no contents. Is that the expected behavior? Or should this be to not create the file at all?

if so you should do this:
```
if caData := string(ign.GetKubeletCAData()); caData != "" {
     	filePathsToContents[windows.K8sDir+"\\"+KubeletClientCAFilename] = string(caData)
}
```

---

I don't feel this is a valid case in the execution flow, given the kubeletCA is [required](https://github.com/openshift/windows-machine-config-operator/pull/1725/files#diff-5f8bc59c6ca4160f53648587fbed538adbd305ad0dfb837bd3124f2ee84fde9bR78).

---

This implies, if for whatever reason, the kubelet CA is empty, we are going to silently fail. How are you handling that?

---

Below the link behind ["required"](https://github.com/openshift/windows-machine-config-operator/pull/1725/files#diff-5f8bc59c6ca4160f53648587fbed538adbd305ad0dfb837bd3124f2ee84fde9bR78) in the above comment where `ignition.New(nc.client)` returns an error. 

https://github.com/openshift/windows-machine-config-operator/pull/1725/files#diff-5f8bc59c6ca4160f53648587fbed538adbd305ad0dfb837bd3124f2ee84fde9bR78

---

OK, that satisfies my concern